plugins/bcli.c

```diff
@@ -1033,61 +1035,52 @@ static void parse_getnetworkinfo_result(struct plugin *p, const char *buf)
 	tal_free(result);
 }
 
+static void log_warm_up(struct plugin *p)
+{
+	plugin_log(p, LOG_UNUSUAL, "Waiting for bitcoind to warm up...");
+}
+
 static void wait_and_check_bitcoind(struct plugin *p)
 {
-	int in, from, status, ret;
+	int in, from, status;
 	pid_t child;
-	const char **cmd = gather_args(bitcoind, "getnetworkinfo", NULL);
-	bool printed = false;
+	const char **cmd =
+	    gather_args(bitcoind, "-rpcwait", "getnetworkinfo", NULL);
 	char *output = NULL;
 
-	for (;;) {
-		tal_free(output);
+	struct timers *timer = tal(cmd, struct timers);
+	timers_init(timer, time_mono());
+	new_reltimer(timer, timer, time_from_sec(BITCOIND_WARMUP_TIMER),
```

Comment:
This won't work, unfortunately.  We'll be stuck in the waitpid call, not in the ioloop.

But the correct way to do this is to also use `rpcwaittimeout`, so it will fail, and we can log a message and retry.

Reply:
(note: rpcwait was supported since bitcoin 0.9, and rpcwaittimeout since 22.0, so we're good here)

Reply:
Just for my understanding, does the timer run synchronously with the event loop, blocking the program until it expires? Or does it run in the background (in a non-blocking manner), calling the function asynchronously, if timer expires or get cleaned up by `tal_free`?